library/psa_crypto.c

```diff
@@ -1327,6 +1350,135 @@ psa_status_t psa_export_key( mbedtls_svc_key_id_t key,
     return( ( status == PSA_SUCCESS ) ? unlock_status : status );
 }
 
+#if defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_FFDH_KEY_PAIR) || \
+    defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_FFDH_PUBLIC_KEY)
+static psa_status_t psa_ffdh_set_prime_generator( size_t key_size,
+                                                  mbedtls_mpi *P,
+                                                  mbedtls_mpi *G )
+{
+    const unsigned char *dhm_P = NULL;
+    const unsigned char *dhm_G = NULL;
+    size_t dhm_size_P = 0;
+    size_t dhm_size_G = 0;
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+
+    if( P == NULL && G == NULL )
+        return( PSA_ERROR_INVALID_ARGUMENT );
+
+    static const unsigned char dhm_P_2048[] =
```

Comment:
We need to keep `P` and `Q` arrays in memory to be able to use them during the computations.
What do you suggest?

Reply:
Ah, right, we're keeping them as macros, I didn't check that. I would then suggest refactoring this a bit, since now, regardless of what key size is used, a lot of the stack space is used for all of the arrays even though only one of them is getting used.

Reply:
No the stack space is not used. The arrays are `static` so they are located in data segment. Only pointer to the suitable `P`, `G` arrays are returned.